tests/upgrades/test_subscription.py

```diff
@@ -162,36 +158,22 @@ def test_post_subscription_scenario_autoattach(
             1. Pre-upgrade content host should get subscribed.
             2. All the cleanup should be completed successfully.
         """
-        docker_vm = settings.upgrade.docker_vm
-        pre_test_name = dependent_scenario_name
-        docker_container_name = f'{pre_test_name}' + "_docker_client"
-        docker_hostname = execute(
-            docker_execute_command,
-            docker_container_name,
-            'hostname',
-            host=docker_vm,
-        )[docker_vm]
-        org = entities.Organization().search(query={'search': f'name="{pre_test_name}_org"'})[0]
-        request.addfinalizer(org.delete)
-        loc = entities.Location().search(query={'search': f'name="{pre_test_name}_loc"'})[0]
-        request.addfinalizer(loc.delete)
-        act_key = entities.ActivationKey(organization=org).search(
-            query={'search': f'name="{pre_test_name}_ak"'}
+        rhel_contenthost = Broker().from_inventory(
+            filter=f'hostname={pre_upgrade_data.rhel_client}'
         )[0]
-        request.addfinalizer(act_key.delete)
-        host = entities.Host(organization=org).search(
-            query={'search': f'name="{docker_hostname.casefold()}"'}
+        host = rhel_contenthost.nailgun_host
+        target_sat.cli.Host.subscription_auto_attach({'host-id': host.id})
+        result = rhel_contenthost.execute('yum install -y zsh')
+        assert result.status == 0, 'package was not installed'
+        result = rhel_contenthost.execute('subscription-manager attach --auto')
+        assert result.status == 0
+        org = target_sat.api.Organization().search(
+            query={'search': f'name={pre_upgrade_data.org_name}'}
         )[0]
-        delete_manifest(org.name)
-        request.addfinalizer(host.delete)
-        subscription = execute(
-            docker_execute_command,
-            docker_container_name,
-            'subscription-manager attach --auto',
-            host=docker_vm,
-        )[docker_vm]
-        assert 'Subscribed' in subscription
-        ssh.command(
-            f"docker stop {docker_container_name}; docker rm {docker_container_name}",
-            hostname=f'{docker_vm}',
-        )
+        request.addfinalizer(org.delete)
+        sub = target_sat.api.Subscription(organization=org)
+        sub.delete_manifest(data={'organization_id': org.id})
+        assert len(sub.search()) == 0
+        manifester = Manifester(manifest_category=settings.manifest.entitlement)
+        manifester.allocation_uuid = pre_upgrade_data.allocation_uuid
+        request.addfinalizer(manifester.delete_subscription_allocation)
```

Comment:
We don't want to use the in-context manager in this case, as it was added specifically for the pre-upgrade part of the testing process. However, we did discuss with @synkd the need for a solution to identify any older subscriptions that were used during testing and need to be cleared on a regular basis by a cron job.

@Gauravtalreja1 One possible solution is to use the pytest_sessionfinish hook to clear these subscriptions in case the post-upgrade process fails. This would be an improvement, but it's not directly related to the current PR.